MIGRATION.md

```diff
@@ -0,0 +1,193 @@
+- `Amount.assertIouIsValid` take a `BigNumber` instead `Decimal`
+- `Amount.verifyNoDecimal` takes a `BigNumber` instead `Decimal`
+
+### Error Handling
+
+In order to remove `assert` as a dependency regular `Error` objects are thrown instead of  `AssertionError`.
+
+Not a breaking change but there is now a much more descriptive error message when keys do not match for a given algorithm.  This impacts most of `ripple-keypairs` functions but only if you already had issues with incompatible values.
+
+### Bundlers
+
+For `vite` and `create-react-app` you can remove all xrpl.js polyfills/configurations this also includes the custom mappings for `ws` to `WsWrapper` and the exclusion of `https-proxy-agent`. These improves should extend to using the libraries with other bundlers or frameworks but has only been tested in those noted above.
+
+#### React Native
+
+Neither `react-native-crypto` nor `rn-nodeify` are required.  Instead `fast-text-encoding` and `react-native-get-random-values` are as well as a small polyfills file to wire them up. You can find the new guide at [UNIQUE_SETUPS.md](./UNIQUE_SETUPS.md).
+
+### `xrpl-secret-numbers`
+
+This library was absorbed into the monorepo as it was causing dependency mismatches and was going to require an update anyway.  It is now available as `@xrplf/secret-numbers`.  Besides making changes to use `UInt8Array` you will need to update all places where you use functions on the `Util` object.  These methods are now at the root of the library. These methods include:
+
+- `Utils.randomEntropy`,
+- `Utils.randomSecret`
+- `Utils.entropyToSecret`
+- `Utils.secretToEntropy`
+- `Utils.calculateChecksum`
+- `Utils.checkChecksum`
+- `Utils.parseSecretString`
+
+##### Prebundle
+
+If you use the pre bundled version of the library you will need to make the following changes:
+
+- Change any references to `dist/browerified.js` to `build/xrplf-secret-numbers-latest.js`.
+- Access any methods as properties of `xrplf_secret_numbers` instead of using browserify's loader.
+
+### HttpsProxyAgent
+
+The way to configure proxies for `Client` has changed. It is now done by specifying the `agent` parameter on the ConnectionOptions config. This can use be created by libraries such as `https-proxy-agent` or any that
+implements `http.Agent`.  This was done to remove a hard dependency on `https-proxy-agent` when running in the browser. The lack of proxy support is not a reduction in features as it never supported running in the browser and merely burdened bundles with unused dependencies.  This was also done to support `https-proxy-agent@7` which changed several option names.
+
+#### Before
+
+```javascript
+{
+  proxy: `ws://127.0.0.1:${port}`,
+  authorization: 'authorization',
+  trustedCertificates: ['path/to/pem'],
+}
+```
+
+#### After
+
+```javascript
+ {
+   agent: new HttpsProxyAgent<string>(`ws://127.0.0.1:${port}`, {
+     ca: ['path/to/pem'],
+   }),
+   authorization: 'authorization'
+ }
```

Comment:
Please include the context in which this would've been used previously (e.g. one line with the Client object having that parameter set).